encoders/nlp/TransformerTorchEncoder/tests/test_transformertorchencoder.py

```diff
@@ -1,140 +1,161 @@
 import os
+
 import numpy as np
 import pytest
-import random
-import string
-from .. import TransformerTorchEncoder
 from jina.executors import BaseExecutor
 from jina.executors.metas import get_default_metas
 
-
-@pytest.fixture(scope='function')
-def random_workspace_name():
-    """Generate a random workspace name with digits and letters."""
-    rand = ''.join(random.choices(string.ascii_uppercase + string.digits, k=6))
-    return f'JINA_TEST_WORKSPACE_{rand}'
+from .. import TransformerTorchEncoder
 
 
 @pytest.fixture(scope='function')
-def test_metas(tmpdir, random_workspace_name):
-    os.environ[random_workspace_name] = str(tmpdir)
+def test_metas(tmp_path):
     metas = get_default_metas()
-    metas['workspace'] = os.environ[random_workspace_name]
+    metas['workspace'] = str(tmp_path)
     if 'JINA_TEST_GPU' in os.environ:
         metas['on_gpu'] = True
     yield metas
-    del os.environ[random_workspace_name]
-
-
-encoders_parameters = [
-    {
-        "pretrained_model_name_or_path": 'sentence-transformers/distilbert-base-nli-stsb-mean-tokens',
-        "model_save_path": 'distilbert-base-nli-stsb-mean-tokens',
-    },
-    {
-        "pooling_strategy": 'auto',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-mean',
-    },
-    {
-        "pooling_strategy": 'min',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-min',
-    },
-    {
-        "pooling_strategy": 'max',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-max',
-    },
-    {
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased',
-    },
-    {
-        "pooling_strategy": 'auto',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-mean',
-    },
-    {
-        "pooling_strategy": 'min',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-min',
-    },
-    {
-        "pooling_strategy": 'max',
-        "pretrained_model_name_or_path": 'xlnet-base-cased',
-        "model_save_path": 'xlnet-base-cased-max',
-    },
-    {
-        "pooling_strategy": 'mean',
-        "pretrained_model_name_or_path": 'distilbert-base-uncased',
-        "model_save_path": 'distilbert-base-uncased-mean',
-        "layer_index": -2,
-    },
-    {
-        "pooling_strategy": 'mean',
-        "pretrained_model_name_or_path": 'distilbert-base-cased',
-        "model_save_path": 'distilbert-base-cased-mean',
-        "max_length": 100
-    },
+
```

Comment:
We need to add a test for checking the quality of emb as there has been a lot of changes. Take emb of the `bert-base-uncased` using the old code and save it as npy. Use it to assert the new code produces same emb. (keep old code defaults such that they match with current)

Reply:
Sure, will do

Reply:
@bhavsarpratik done, tests pass :)